File: DSA/88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int l=0, r=0;
        vector<int> ans = nums1;
        int i = 0;
        while(l<m && r<n){
            if(ans[l] < nums2[r]){
                nums1[i] = ans[l];
                ++l;
            }
            else{
                nums1[i] = nums2[r];
                ++r;
            }
            ++i;
        }
        for(int j=r; j<n; j++){
            nums1[i] = nums2[j];
            ++i;
        }
        for(int j=l; j<m; j++){
            nums1[i] = ans[j];
            ++i;
        }
    }      
};

File: DSA/88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        if n == 0: return 
        end_i = len(nums1)-1
        while n> 0 and m>0:
            if nums2[n-1] >= nums1[m-1]:
                nums1[end_i] = nums2[n-1]
                n-=1
            else:
                nums1[end_i]= nums1[m-1]
                m-=1
            end_i-=1
    
        while n>0:
            nums1[end_i] = nums2[n-1]
            n-=1
            end_i -=1


